etc/compile_check.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+set -e # exit when any command fails
+
+GC=go
+WD=$(pwd)
+COMPILE_CHECK_DIR="internal/test/compilecheck"
+DEV_MIN_VERSION=1.19
+
+# version will flatten a version string of upto 4 components for inequality
+# comparison.
+function version {
+	echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }';
+}
+
+# dev_compile_check will attempt to build in the development environment. This
+# check will only run on environments where the Go version is greater than or
+# equal to the DEV_MIN_VERSION.
+function dev_compile_check {
+	MACHINE_VERSION=`${GC} version | { read _ _ v _; echo ${v#go}; }`
+
+	if [ $(version $MACHINE_VERSION) -lt $(version $DEV_MIN_VERSION) ]; then
+		return
+	fi
+
+	GO111MODULE=on ${GC} build ./...
+	GO111MODULE=on ${GC} build ${BUILD_TAGS} ./...
```

Comment:
The `GO111MODULE` behavior is enabled by default on Go 1.13+ if there is a `go.mod` file (see [details](https://go.dev/doc/go1.13#modules)). Is there a reason we need to explicitly enable it?